txcache/virtualSelectionSession.go

```diff
@@ -25,26 +26,50 @@ func (virtualSession *virtualSelectionSession) getRecord(address []byte) (*virtu
 		return virtualRecord, nil
 	}
 
-	account, err := virtualSession.session.GetAccountState(address)
+	virtualRecord, err := virtualSession.createAccountRecord(address)
 	if err != nil {
-		log.Debug("virtualSelectionSession.getRecord",
+		log.Debug("virtualSelectionSession.getRecord: error when creating virtual account record",
 			"address", address,
 			"err", err)
 		return nil, err
 	}
 
+	// We handle records corresponding to new (missing) accounts, as well (see "createAccountRecord").
+	virtualSession.virtualAccountsByAddress[string(address)] = virtualRecord
+	return virtualRecord, nil
+}
+
+func (virtualSession *virtualSelectionSession) createAccountRecord(address []byte) (*virtualAccountRecord, error) {
+	account, err := virtualSession.session.GetAccountState(address)
+	if err != nil {
+		if err == state.ErrAccNotFound {
```

Comment:
perhps extract this if err == state.ErrAccNotFound before if err != nil

Reply:
Fixed.